server/src/test/java/org/elasticsearch/common/xcontent/support/XContentHelperTests.java

```diff
@@ -230,4 +233,12 @@ public void testEmptyChildBytes() throws IOException {
         }
 
     }
+
+    public void testConvertToMapYaml() {
```

Comment:
Can we have some JavaDoc here that explains that the YAML in the test doesn't include the `---` preamble, but because the test calls `convertToMap` in non-strict mode, everything still works?

Maybe we can name the test `testConvertYamlToMapInNonStrictMode`?

Personally speaking, I'd be tempted to include tests for strict-mode, both with and without the `---` header, and for non-strict mode with the header.